pkg/conn/basedb.go

```diff
@@ -80,14 +89,26 @@ func (d *BaseDB) GetBaseConn(ctx context.Context) (*BaseConn, error) {
 	if err != nil {
 		return nil, terror.ErrDBDriverError.Delegate(err)
 	}
-
-	return newBaseConn(conn, d.Retry), nil
+	baseConn := newBaseConn(conn, d.Retry)
+	d.conns = append(d.conns, baseConn)
```

Comment:
2 questions:

1. when resetting `BaseConn`, should we need to close the previous `BaseConn` firstly?
2. `d.conns = append(d.conns, baseConn)` seems not thread-safe, is there any problem?

Reply:
1. Yes, we should close, I'll fix it 
2. In syncer toDBConns, each connection is independent, but if they need to reset at same time then get a problem, I'll fix it later